checkov/terraform/graph_builder/local_graph.py

```diff
@@ -382,10 +399,15 @@ def _get_dest_module_path(self, curr_module_dir: str, dest_module_source: str, d
         return os.path.abspath(dest_module_path)
 
     def _find_vertex_index_relative_to_path(
-        self, block_type: BlockType, name: str, block_path: str, module_path: str, module_num: str, relative_module_idx: Optional[int] = None
+        self, block_type: BlockType, name: str, block_path: str, module_path: str, module_num: str, relative_module_idx: Optional[int] = None, source_module_object: Optional[TFModule] = None
     ) -> int:
         relative_vertices = []
-        if relative_module_idx is not None:
+        if self.use_new_tf_parser:
+            if relative_module_idx is None:
+                module_dependency_by_name_key = source_module_object
+            else:
+                module_dependency_by_name_key = next(k for k, v in self.vertices_by_module_dependency.items() if v.get(BlockType.MODULE, []).__contains__(relative_module_idx))
+        elif relative_module_idx is not None:
             module_dependency_by_name_key = next(k for k, v in self.vertices_by_module_dependency.items() if v.get(BlockType.MODULE, []).__contains__(relative_module_idx))
```

Comment:
```suggestion
        if self.use_new_tf_parser and relative_module_idx is None:
            module_dependency_by_name_key = source_module_object
        elif relative_module_idx is not None:
            module_dependency_by_name_key = next(k for k, v in self.vertices_by_module_dependency.items() if v.get(BlockType.MODULE, []).__contains__(relative_module_idx))
```
shouldn't this be the same?